notifier/service.go

```diff
@@ -44,6 +44,17 @@ func (s *Service) Notify(lastChange time.Time, transactionID string) error {
 	if err != nil {
 		return fmt.Errorf("failed to fetch the list of changed concepts: %w", err)
 	}
+
+	if len(changedConcepts) == 0 {
+		// After some time interval retry getting the changed concept list,
+		// because Smartlogic could have notified us before the data is available to be retrieved.
```

Comment:
I suggest I slight rewording to be clear that this is not a theoretical case.
```suggestion
		// because Smartlogic sometimes notify us before the data is available to be retrieved.
```